apps/openmw/mwmechanics/aiwander.cpp

```diff
@@ -303,8 +334,55 @@ namespace MWMechanics
 
             if (storage.mPathFinder.isPathConstructed())
             {
-                storage.mState = Wander_Walking;
+                storage.setState(Wander_Walking);
+            }
+        }
+    }
+
+    /*
+     * Commands actor to walk to a random location near original spawn location.
+     *
+     * Creatures simply wander a certain distance from their starting location, while NPCs wander a scripted
+     * distance (mDistance) from the position where they started the wander package.
+     * http://www.uesp.net/wiki/Tes3Mod:AIWander
+     */
+    void AiWander::wanderNearStart(const MWWorld::Ptr &actor, AiWanderStorage &storage, int wanderDistance, bool isNpc) {
+        const ESM::Pathgrid::Point currentPosition = actor.getRefData().getPosition().pos;
+
+        // Determine a random location within radius of original position
+        const float pi = 3.14159265359f;
+        const float maxWanderDistance = isNpc ? wanderDistance : MINIMUM_WANDER_DISTANCE * 14.0f;
```

Comment:
Are you sure that this is correct? From my testing, creatures also respect the wander distance given in their AI package. For example, giving a very small distance: 
<pre>
aiwander, 20, 0, 0
</pre>
a scrib would become stationary.

Given a very high distance:
<pre>
aiwander, 10000, 0, 0
</pre>
it would keep walking in one direction until it hit something.